R/expr-expr.R

```diff
@@ -1070,102 +1070,117 @@ expr__filter <- function(...) {
     wrap()
 }
 
-## TODO Better explain aggregate list
-
-#' Map an expression with an R function
+#' Apply a custom R function to a whole Series or sequence of Series.
 #'
-#' @inheritParams rlang::args_dots_empty
-#' @param f a function to map with
-#' @param output_type `NULL` or a type available in `names(pl$dtypes)`. If `NULL`
-#' (default), the output datatype will match the input datatype. This is used
-#' to inform schema of the actual return type of the R function. Setting this wrong
-#' could theoretically have some downstream implications to the query.
-#' @param agg_list Aggregate list. Map from vector to group in group_by context.
-#' @param in_background Logical. Whether to execute the map in a background R
-#' process. Combined with setting e.g. `options(polars.rpool_cap = 4)` it can speed
-#' up some slow R functions as they can run in parallel R sessions. The
-#' communication speed between processes is quite slower than between threads.
-#' This will likely only give a speed-up in a "low IO - high CPU" use case.
-#' If there are multiple [`$map_batches(in_background = TRUE)`][expr__map_batches]
-#' calls in the query, they will be run in parallel.
+#' @description
+#' The output of this custom function is presumed to be either a Series, or a
+#' scalar that will be converted into a Series. If the result is a scalar and
+#' you want it to stay as a scalar, pass in `returns_scalar = TRUE`.
 #'
+#' If you want to apply a custom function elementwise over single values, see
+#' [map_elements()][expr__map_elements]. A reasonable use case for map
+#' functions is transforming the values represented by an expression using a
+#' third-party package.
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @param f Function to apply.
```

Comment:
Param named `function` in py-polars but it's a keyword in R

Reply:
I renamed to `lambda`, it's not like R is consistent on this (sometimes `FUN`, sometimes `f` for function, sometimes `f` for factors, etc.)